beacon_node/beacon_chain/src/data_availability_checker/overflow_lru_cache.rs

```diff
@@ -600,13 +615,18 @@ impl<T: BeaconChainTypes> OverflowLRUCache<T> {
 
         // Grab existing entry or create a new entry.
         let mut pending_components = write_lock
-            .pop_pending_components(block_root, &self.overflow_store)?
+            .get_pending_components(block_root, &self.overflow_store)?
             .unwrap_or_else(|| PendingComponents::empty(block_root));
 
         // Merge in the blobs.
         pending_components.merge_blobs(fixed_blobs);
 
         if pending_components.is_available() {
+            write_lock.put_pending_components(
+                block_root,
+                pending_components.clone(),
+                &self.overflow_store,
+            )?;
```

Comment:
We're cloning pending components for a second time here and I'm not sure if it's necessary.
If the pending component is newly created in this function, it won't immediately become available. So this means the pending components would always be in the `overflow_store` if it becomes available here right?

Reply:
Oh never mind, we do need this as we're mutating a clone!

Reply:
You don't need to modify `pop_pending_components` to `get_pending_components`. Instead you should keep it as is. Because the write lock is held the whole time, there is no difference between:
```rust
lock = get_write_lock()
item = cache.remove()
item.mutate()
cache.insert(item)
drop(lock)
```

```rust
lock = get_write_lock()
item = cache.get()
item.mutate()
cache.update(item)
drop(lock)
```